lib/analyzer/applications/python/poetry.ts

```diff
@@ -0,0 +1,89 @@
+import { DepGraph } from "@snyk/dep-graph";
+import * as path from "path";
+import * as lockFileParser from "snyk-poetry-lockfile-parser";
+import { DepGraphFact, TestedFilesFact } from "../../../facts";
+import { AppDepsScanResultWithoutTarget, FilePathToContent } from "../types";
+
+interface ManifestLockPathPair {
+  manifest: string;
+  lock: string;
+}
+
+export async function poetryFilesToScannedProjects(
+  filePathToContent: FilePathToContent,
+): Promise<AppDepsScanResultWithoutTarget[]> {
+  const scanResults: AppDepsScanResultWithoutTarget[] = [];
+
+  const filePairs = findManifestLockPairsInSameDirectory(filePathToContent);
+
+  const shouldIncludeDevDependencies = false;
+
+  for (const pathPair of filePairs) {
+    const depGraph = await lockFileParser.buildDepGraph(
+      filePathToContent[pathPair.manifest],
+      filePathToContent[pathPair.lock],
+      shouldIncludeDevDependencies,
+    );
+
+    const depGraphFact: DepGraphFact = {
+      type: "depGraph",
+      data: depGraph as DepGraph,
```

Comment:
Same as above - can this be undefined and is it ok :)

Reply:
good catch. here I should add a check before like in the `pip` analyzer